nativelib/src/main/scala/scala/scalanative/runtime/Intrinsics.scala

```diff
@@ -134,9 +139,63 @@ object Intrinsics {
   /** Intrinsified cast that reinterprets raw pointer as an long. */
   def castRawPtrToLong(rawptr: RawPtr): Long = intrinsic
 
+  /** Intrinsified cast that reinterprets raw pointer as a raw size. */
+  def castRawPtrToRawSize(rawptr: RawPtr): RawSize = intrinsic
+
   /** Intrinsified cast that reinterprets int as a raw pointer. */
   def castIntToRawPtr(int: Int): RawPtr = intrinsic
 
   /** Intrinsified cast that reinterprets long as a raw pointer. */
   def castLongToRawPtr(int: Long): RawPtr = intrinsic
+
+  /** Intrinsified cast that reinterprets raw size as an int. */
+  def castRawSizeToInt(rawSize: RawSize): Int = intrinsic
+
+  /** Intrinsified cast that reinterprets raw size as a signed long. */
+  def castRawSizeToLong(rawSize: RawSize): Long = intrinsic
```

Comment:
I don't think we actually need bitwise and math operations on `RawSize`. In case if we need that we can always cast it from int/long as it's done with `RawPtr`.

Reply:
IIRC, we can replace these intrinsics by casting both operands to `Int` or `Long` depending on the target architecture and then casting back? I guess as long as the conditionals are appropriately collapsed at link time the performance would be the same.

Reply:
Fixed! But we should probably do a performance comparison just to make things are optimized appropriately.